packages/wrangler/e2e/unenv-preset/worker/index.ts

```diff
@@ -562,3 +562,57 @@ export const WorkerdTests: Record<string, () => void> = {
 		);
 	},
 };
+
+/**
+ * Creates helper functions to test the properties of an imported module and a global object.
+ *
+ * Depending upon the compat flags and date, the namespace and global may be native or polyfilled.
+ * We can't infer the imported namespace from the import specifier because the bundler would treat it as dynamic
+ * and not provide the polyfill.
+ *
+ * @param importSpecifier The module specifier of the module being tested.
+ * @param importNamespace The imported module namespace object.
+ * @param globalProperty The property of the global object to test.
+ * @param globalObject The global object being tested.
+ * @returns A set of helper functions for testing the properties of the imported module and global object.
```

Comment:
As discussed on the other PR, `testPropertyType` is not really self-describing and IMO we should favor inlining in the tests.

One helper that could be self-describing could be:

```ts
assertPropertyType(target, "path.to.prop", type);
```

`"path.to.prop"` can be re-used in the error message to get better error in case of a failed assertion.

We could also `for of` this helper